### DISCUSSION TITLE
New Version Parse failed

### DISCUSSION THREAD

**Discussion by itisyxfan** [sourcelair/xterm.js#5201] [Q&A]
This is normal when using an older version of xterm. Just like this:
```js
yarn add xterm
import { Terminal } from 'xterm';
```
But the new version of xterm fails to parse. Just like this:
```js
yarn add @xterm/xterm
 import { Terminal } from '@xterm/xterm';
```
The error message is as follows：
```js
 error  in ./node_modules/@xterm/xterm/lib/xterm.js
Module parse failed: Unexpected token (1:4232)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
(Source code omitted for this binary file)
```
Please help me, it would be much appreciated

**Comment by jerch** [comment]
Looks like a webpack config issue to me.

**Reply by itisyxfan** [reply]
But it is possible to use xterm directly. Is there any difference between xterm and @xterm/xterm? Looking forward to your reply

**Reply by jerch** [reply]
The ESM packaging is not yet in @xterm/xterm, it got recently added and is still in beta, so package-wise xterm is the same as @xterm/xterm.
My guess here is, that webpack assumes an ES module, but in fact it is not yet. So maybe downgrade webpack (or find an appropriate switch to turn that behavior off) or maybe try the beta releases with real ESM?

Edit: In svelte I've seen a different workaround by importing the default namespace like this:
```javascript
import xterm from '@xterm/xterm';
...
const term = xterm.Terminal(...);
```
Maybe this works here too, idk.

**Reply by jerch** [reply]
Btw I just tested it with an empty webpack config and `import { Terminal } from '@xterm/xterm';` in index.js - works for me even with newest webpack, so this is def. a webpack config issue on your side.

**Reply by itisyxfan** [reply]
May I ask what is your webpack version？My version is v4.46.0

**Reply by jerch** [reply]
package.json says:
```
  "dependencies": {
    "@xterm/xterm": "^5.5.0",
    "webpack": "^5.95.0"
  },
  "devDependencies": {
    "webpack-cli": "^5.1.4"
  }
```

**Reply by itisyxfan** [reply]
Thanks for your reply. I just tried webpack5 and it worked fine. Since webpack4 is used in the project, I'll use xterm instead of @xterm/xterm for now.

**Reply by jerch** [reply]
Oh thats wild. Well with just `xterm` you are basically on v5.3 and miss out the changes up to v5.5. Plz see the [releases](https://github.com/xtermjs/xterm.js/releases) and whether they contain relevant changes for you.
